.github/workflows/macos_ui_tests.yml

```diff
@@ -28,7 +42,7 @@ on:
       - '.xcode-version'
 
 concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}
+  group: ${{ github.workflow }}-${{ github.ref }}-ui-tests
```

Comment:
When this workflow is called by another workflow, github.workflow contains the name of the calling workflow, and this (old value) would be equal for this and pr-checks, which would prevent them from running concurrently. I’m adding a suffix to the concurrency group to tell them apart and allow for running PR Checks and UI Tests side by side in the code freeze workflow.